source/elements/oneDAL/source/data_management/index.rst

```diff
@@ -126,113 +130,93 @@ used at the data preparation and data processing stages for the following:
 - To transfer memory ownership of the data from user application to the table,
   or share it between them.
 
-- To connect with the :txtref:`data-source` to convert from an out-of-memory
-  into an in-memory dataset representation.
+- To connect with the :txtref:`data-source` to convert data from an
+  out-of-memory into an in-memory representation.
 
 - To support streaming of the data to the algorithm.
 
 - To access the underlying data on a device in a required :capterm:`data
   format`, e.g. by blocks with the defined :capterm:`data layout`.
 
-For thread-safety reasons and better integration with external entities, a table
-provides a read-only access to the data within it, thus, table concept
-implementations shall be :capterm:`immutable <immutability>`.
+.. note::
+  For thread-safety reasons and better integration with external entities, a table
+  provides a read-only access to the data within it, thus, table concept
+  implementations shall be :capterm:`immutable <immutability>`.
+
+.. warning::
+  Table can represent not a datasets only - it can be a matrix, a vector (if
+  number of columns :math:`p = 1`), even a single value.
+
+  When table represent a dataset, the table rows represent dataset observations,
+  and columns represent features according to the :txtref:`dataset` definition.
 
 This concept has different logical organization and physical :capterm:`format of
 the data <data format>`:
 
-- Logically, a table is a :txtref:`dataset` with :math:`n` rows and
-  :math:`p` columns. Each row represents an :capterm:`observation` and each
-  column is a :capterm:`feature` of a dataset. Physical amount of bytes needed
-  to store the data differ from the number of elements :math:`n \times p` within
-  a table.
+- Logically, a table contains :math:`n` rows and :math:`p` columns.
+  Every column may have its own type of data values and set of allowed
+  operations on them.
 
 - Physically, a table can be organized in different ways: as a
   :capterm:`homogeneous <homogeneous data>`, :capterm:`contiguous <contiguous
   data>` array of bytes, as a :capterm:`heterogeneous <heterogeneous data>` list
   of arrays of different :capterm:`data types <data type>`, in a
-  compressed-sparse-row format.
+  compressed-sparse-row format. Physical amount of bytes needed to store the
```

Comment:
not sure 'physical amount of bytes' would be clean. @outoftardis, pls suggest a better wording

Reply:
What about "the number of bytes in physical memory"?